crates/turbopack-core/src/resolve/pattern.rs

```diff
@@ -627,6 +627,33 @@ impl Pattern {
         new.normalize();
         Pattern::alternatives([self.clone(), new])
     }
+
+    /// Calls `cb` on all constants that are at the end of the pattern and
+    /// replaces the given final constant with the returned pattern. Returns
+    /// true if replacements were performed.
+    pub fn replace_final_constants(&mut self, cb: &impl Fn(&RcStr) -> Option<Pattern>) -> bool {
+        let mut replaced = false;
+        match self {
+            Pattern::Constant(c) => {
+                if let Some(replacement) = cb(c) {
+                    *self = replacement;
+                    replaced = true;
+                }
+            }
+            Pattern::Dynamic => {}
+            Pattern::Alternatives(list) => {
+                for i in list {
+                    replaced = i.replace_final_constants(cb) || replaced;
+                }
+            }
+            Pattern::Concatenation(list) => {
+                if let Some(c @ Pattern::Constant(_)) = list.last_mut() {
```

Comment:
The check for constant should be removed here as the last element might be something complex too